### ISSUE TITLE
Latex mode freeze

### ISSUE THREAD

**Issue by vikke1234** [syl20bnr/spacemacs#14342]
#### Description :octocat:
Freezing with latex layer and texlab

#### Reproduction guide :beetle:
- Start Emacs
- open tex file (texlab in PATH)
- spend ~5 seconds in it

*Observed behaviour:* :eyes: :broken_heart:
Completely freezes not willing to close without task manager

NOTE: it doesn't freeze if you don't have texlab in PATH which means it has something to do with it.

*Expected behaviour:* :heart: :smile:
Freezes completely

#### System Info :computer:
- OS: windows-nt
- Emacs: 27.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. a65497be7)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(html autohotkey
      (javascript :variables javascript-import-tool 'import-js javascript-backend 'lsp javascript-fmt-tool 'prettier)
      react python windows-scripts restclient emacs-lisp
      (auto-completion :variables auto-completion-enable-snippets-in-popup t)
      (cmake :variables cmake-backend 'lsp cmake-enable-cmake-ide-support t)
      (tabs :variables centaur-tabs-style "rounded")
      git helm lsp
      (markdown :variables markdown-live-preview-engine 'vmd)
      multiple-cursors org latex asm
      (c-c++ :variables c-c++-enable-clang-support t c-c++-enable-organize-includes-on-save t c-c++-backend 'lsp-clangd)
      (shell :variables shell-default-height 30 shell-default-position 'bottom)
      syntax-checking version-control
      (treemacs :variables treemacs-filewatch-mode t))
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY W32NOTIFY ACL GNUTLS LIBXML2 HARFBUZZ ZLIB TOOLKIT_SCROLL_BARS MODULES THREADS JSON PDUMPER LCMS2 GMP

#### Backtrace :paw_prints:
N/A since it just freezes...

**Comment by FilMarini** [comment]
Happens to me as well only with lsp as backend. Its impossible to use.

I'm using Ubuntu 18.04

**Comment by lebensterben** [comment]
LSP doesn't work well with some unicode symbols.
For example, in any mode with LSP enabled, and if you also enabled ligatures, it freezes indefinitely.

So first we need to make sure the observed behavior is not caused by unicode.

**Comment by alexey0308** [comment]
Probably relevant https://github.com/ROCKTAKEY/lsp-latex/issues/19  https://github.com/latex-lsp/texlab/issues/284

**Comment by arifer612** [comment]
I found [a very relevant issue](https://github.com/mickeynp/ligature.el/issues/36) upstream that seems to explain the gruesome detail as to why this may be the case for my situation using Emacs 29.0.50. I believe that the necessary fixes seem to have been implemented and I can comfortably say that my massive Python projects can be worked on with full ligatures without the jerky freezes anymore with the latest updates.

If you want ligatures to work in Emacs, I suggest updating it to at least 28.1 and try to see if there's a significant improvement in the response.
